stream/src/main/java/org/apache/kafka/streaming/kstream/internals/KStreamJoin.java

```diff
@@ -122,6 +129,34 @@ public void process(K key, V1 value) {
             }
         }
 
+        private boolean joinable() {
+            Set<TopicPartition> partitions = ((ProcessorContextImpl) this.context).task().partitions();
```

Comment:
KStreamJoin should not care how the joinability is computed. So, having joinable() in ProcessorContext is actually good.

Anything that makes tests harder is bad since it is an indication of a bad code organization.